### PR TITLE
webapi : Reset user rating command

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

File: src/Application/Characters/Commands/ResetCharacterRatingCommand.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            _mapper = mapper;
+            _characterService = characterService;
+            _activityLogService = activityLogService;
+        }
+
+        public async Task<Result<CharacterViewModel>> Handle(ResetCharacterRatingCommand req, CancellationToken cancellationToken)
+        {
+            var character = await _db.Characters
+                .Include(c => c.User)
+                .Include(c => c.EquippedItems)
```
Comment (verdie-g) [inline src/Application/Characters/Commands/ResetCharacterRatingCommand.cs:40]: Not needed

File: src/Application/Common/Services/IActivityLogService.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ internal interface IActivityLogService
     ActivityLog CreateItemUpgradedLog(int userId, string itemId, int heirloomPoints);
     ActivityLog CreateCharacterCreatedLog(int userId, int characterId);
     ActivityLog CreateCharacterDeletedLog(int userId, int characterId, int generation, int level);
+    ActivityLog CreateCharacterRatingResetedLog(int userId, int characterId);
```
Comment (verdie-g) [inline src/Application/Common/Services/IActivityLogService.cs:19]: The past participle of reset is reset

File: src/WebApi/Controllers/UsersController.cs [hunk 3]
```diff
@@ -356,6 +356,18 @@ public Task<ActionResult<Result<CharacterLimitationsViewModel>>> GetCharacterLim
         }));
     }
 
+    /// <summary>
+    /// Resets a character Rating.
+    /// </summary>
+    /// <param name="userId">Owner of Character userId.</param>
```
Comment (verdie-g) [inline src/WebApi/Controllers/UsersController.cs:362]:
```suggestion
    /// Resets a character rating.
    /// </summary>
    /// <param name="userId">User id of the character owner.</param>
```

File: src/WebApi/Controllers/UsersController.cs [hunk 4]
```diff
@@ -356,6 +356,18 @@ public Task<ActionResult<Result<CharacterLimitationsViewModel>>> GetCharacterLim
...
+    /// <param name="id">Character id.</param>
+    /// <response code="200">Character rating reseted.</response>
```
Comment (verdie-g) [inline src/WebApi/Controllers/UsersController.cs:364]:
```suggestion
    /// <response code="200">Character rating reset.</response>
```

File: src/WebApi/Controllers/UsersController.cs [hunk 5]
```diff
@@ -356,6 +356,18 @@ public Task<ActionResult<Result<CharacterLimitationsViewModel>>> GetCharacterLim
...
+    /// <response code="400">Bad Request.</response>
+    /// <response code="404">Character not found.</response>
+    [HttpPut("{userId}/characters/{id}/retire")]
```
Comment (verdie-g) [inline src/WebApi/Controllers/UsersController.cs:367]: This endpoint should be restricted to admins only

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Application/Characters/Commands/ResetCharacterRatingCommand.cs (hunk 1 above)
Comment (namidaka) [reply src/Application/Characters/Commands/ResetCharacterRatingCommand.cs]: done

File: src/WebApi/Controllers/UsersController.cs (hunk 4 above)
Comment (namidaka) [reply src/WebApi/Controllers/UsersController.cs]: done

File: src/WebApi/Controllers/UsersController.cs (hunk 5 above)
Comment (namidaka) [reply src/WebApi/Controllers/UsersController.cs:368]: done

**Review by verdie-g — APPROVED** [review]

**Pull request merged**